File: benchmarks/C++HJ/richards/WorkerTaskDataRecord.cpp
#include "RBObject.h"

namespace richards {
    class WorkerTaskDataRecord : public  RBObject {
        private:
            int _destination;
            int _count;

        public:
            WorkerTaskDataRecord() {
                _destination = HANDLER_A;
                _count = 0;
            }
                
            int getCount() const {
                return _count; 
            }

            void setCount(int aCount) { 
                _count = aCount; 
            }

            int getDestination() const {
                return _destination; 
            }

            void setDestination(int aHandler) { 
                _destination = aHandler; 
            }
    };
}

File: benchmarks/Rust/src/richards/worker_task_data_record.rs
use crate::richards::rb_object::RBObject;

pub struct WorkerTaskDataRecord {
    destination: usize,
    count: usize,
}

impl Default for WorkerTaskDataRecord {
    fn default() -> Self {
        Self {
            destination: Self::HANDLER_A,
            count: 0,
        }
    }
}

impl RBObject for WorkerTaskDataRecord {}

impl WorkerTaskDataRecord {
    pub fn get_count(&self) -> usize {
        self.count
    }

    pub fn set_count(&mut self, count: usize) {
        self.count = count;
    }

    pub fn get_destination(&self) -> usize {
        self.destination
    }

    pub fn set_destination(&mut self, destination: usize) {
        self.destination = destination;
    }
}
